### PR TITLE
Add API key support for docker

### PR DESCRIPTION
**Pull request by h0ngcha0**

### CODE REVIEW

File: api/src/main/scala/org/alephium/api/BaseEndpoint.scala [hunk 1]
```diff
@@ -36,18 +36,26 @@ trait BaseEndpoint extends ErrorExamples with TapirCodecs with TapirSchemasLike
...
+    .in(auth.apiKey(header[Option[ApiKey]]("X-API-KEY")))
+
+  val baseEndpointWithoutApiKey: BaseEndpoint[Unit, Unit] =
+    baseEndpointCommon
+      .serverLogicForCurrent { _ =>
+        Future.successful(Right(()): Either[ApiError[_ <: StatusCode], Unit])
+      }
+
+  val baseEndpoint: BaseEndpoint[Unit, Unit] =
+    baseEndpointCommon
```
Comment (polarker) [inline api/src/main/scala/org/alephium/api/BaseEndpoint.scala:58]: Can we simplify it as the following ?
```
  val baseEndpointWithoutApiKey: BaseEndpoint[Unit, Unit] =
    endpoint
      .errorOut(
        oneOf[ApiError[_ <: StatusCode]](
          error(BadRequest, { case BadRequest(_) => true }),
          error(InternalServerError, { case InternalServerError(_) => true }),
          error(NotFound, { case NotFound(_) => true }),
          error(ServiceUnavailable, { case ServiceUnavailable(_) => true }),
          error(Unauthorized, { case Unauthorized(_) => true })
        )
      )
  val baseEndpoint = baseEndpointWithoutApiKey
    .in(auth.apiKey(header[Option[ApiKey]]("X-API-KEY")))
    .serverLogicForCurrent { apiKey => Future.successful(checkApiKey(apiKey)) }

```

File: docker/README.md [hunk 2]
```diff
@@ -23,12 +23,18 @@ The provided [docker-compose.yml](./docker-compose.yml) file will be used to run
 docker-compose stop && docker-compose rm -f alephium && docker-compose pull && docker-compose up -d
 ```
 
-The default config connects your container to the mainnet, and makes the API available to [http://127.0.0.1:12973/docs](http://127.0.0.1:12973/docs):
+The default config connects your container to the mainnet, and makes the API available to [http://127.0.0.1:12973/docs](http://127.0.0.1:12973/docs).
+
+By default an API key with the value of `0000000000000000000000000000000000000000000000000000000000000000` is configured to protect the APIs exposed by
```
Comment (polarker) [inline docker/README.md:28]: could you mentioned that API key should be >= 32 alphabetic + number characters

**Review by polarker — COMMENTED** [review]
Looks great, just two minor comments

File: api/src/main/scala/org/alephium/api/BaseEndpoint.scala (hunk 1 above)
Comment (h0ngcha0) [reply api/src/main/scala/org/alephium/api/BaseEndpoint.scala]: Much better, thanks!

File: docker/README.md (hunk 2 above)
Comment (h0ngcha0) [reply docker/README.md:28]: Good idea, done :+1:

**Review by polarker — APPROVED** [review]

**Pull request merged**